crates/wasmparser/src/validator.rs

```diff
@@ -836,9 +836,7 @@ impl Validator {
     ///
     /// This method should only be called when parsing a module.
     pub fn data_section(&mut self, section: &crate::DataSectionReader<'_>) -> Result<()> {
-        let mut section = section.clone();
-        section.forbid_bulk_memory(!self.features.bulk_memory);
-
+        let section = section.clone();
```

Comment:
I think this may no longer need cloning now that the bulk memory feature is gone